CONTRIBUTING.RELEASE.md

```diff
@@ -1,30 +1,39 @@
 # OpenTofu release manual
 
 > [!WARNING]
-> This manual is intended for OpenTofu core and fork maintainers. If you are looking for the normal contribution guide, see [this file](CONTRIBUTING.md).
+> This manual is intended for OpenTofu maintainers. If you are looking for the normal contribution guide, refer to [`CONTRIBUTING.md`](CONTRIBUTING.md).
```

Comment:
We should just rename this file to RELEASE.md or similar.  I don't know why it was originally named CONTRIBUTING.RELEASE.

Reply:
/facepalm, you are right.  Either way, if we can simplify the release process enough most of this goes away anyways.